common/src/main/java/com/wynntils/handlers/chat/ChatHandler.java

```diff
@@ -111,7 +113,7 @@ public void onChatReceived(ChatPacketReceivedEvent e) {
                 && codedMessage.getInternalCodedStringRepresentation().indexOf('\n')
                         != (codedMessage.getInternalCodedStringRepresentation().length() - 1)) {
             // This is a "chat screen"
-            if (shouldSeparateNPC()) {
+            if (shouldSeparateNPC() || isWelcomeMessage(codedMessage)) {
```

Comment:
I am afraid of touching chat handler this way, I don't think this was intended to be used this way. Do we really need to do this now? I would rather not filter these messages. Also, these messages are Wynncraft's self-promo, and most likely we should keep them

Reply:
I am pretty sure this bug existed for as long as these features also existed. There is no other way for the MessageFilter Feature to get the event that it actually filters, and as long as you had NPC Dialogue disabled, that event would never be posted for multiline messages.

The welcome message is a multiline message, so it would never have been filtered with NPC Dialogue disabled.

Also, I only added filters for the non-self-promo stuff, like the "Welcome to Wynncraft!" and the website link/ip. The event promos like the plushie stuff still exists.